Tests/Admin/AdminTestCase.php

```diff
@@ -0,0 +1,156 @@
+use Sonata\AdminBundle\Builder\ListBuilderInterface;
+use Sonata\AdminBundle\Builder\RouteBuilderInterface;
+use Sonata\AdminBundle\Builder\ShowBuilderInterface;
+use Sonata\AdminBundle\Route\RouteGeneratorInterface;
+use Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface;
+use Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface;
+use Sonata\DoctrineORMAdminBundle\Model\ModelManager;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * Class AdminTestCase
+ *
+ * @author Sylvain Rascar <sylvain.rascar@ekino.com>
+ */
+abstract class AdminTestCase extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ModelManager|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $modelManagerMock;
+
+    /**
+     * @var FormContractorInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $formContractorMock;
+
+    /**
+     * @var ShowBuilderInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $showBuilderMock;
+
+    /**
+     * @var ListBuilderInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $listBuilderMock;
+
+    /**
+     * @var DatagridBuilderInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $datagridBuilderMock;
+
+    /**
+     * @var TranslatorInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $translatorMock;
+
+    /**
+     * @var Pool|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $poolMock;
+
+    /**
+     * @var RouteGeneratorInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $routeGeneratorMock;
+
+    /**
+     * @var ValidatorInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $validatorMock;
+
+    /**
+     * @var SecurityHandlerInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $securityHandlerMock;
+
+    /**
+     * @var FactoryInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $menuFactoryMock;
+
+    /**
+     * @var RouteBuilderInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $routeBuilderMock;
+
+    /**
+     * @var LabelTranslatorStrategyInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $labelTranslatorStrategyMock;
+
+    /**
+     * Default dummy variables
+     *
+     * @var string
+     */
+    protected $dummyAdminId = 'sonata.admin';
+    protected $dummyController = 'SonataAdminBundle:CRUD';
+
+    /**
+     * Create mocks of all admin default services
+     * and assign them to a property of the class
+     *
+     * @param AdminInterface $admin
+     */
+    protected function mockDefaultServices(AdminInterface $admin)
+    {
+        // Each element is composed of: original_service_key, propertyName, class
+        $defaultServices = [
+            ['model_manager', 'modelManagerMock', ModelManager::class],
```

Comment:
Might be interesting to have this configurable (if we need to override one of the classes for instance)

Reply:
I don't want it configurable. This function is supposed to simulate the compiler pass of sonata. That's also why `applyDefaults` is private
If you want to override a class, you have access to the property and you have access to the admin to call it's setter

Reply:
Ok, and how would you handle adding another mocked service to the admin? Might be worth enhancing the documentation with a use case for that, if this doesn't allow it.

Reply:
Done